src/error.rs

```diff
@@ -63,6 +63,9 @@ pub enum ContractError {
     #[error("Total (price * size) must be an integer")]
     NonIntegerTotal,
 
+    #[error("Integer overflow")]
+    OverflowError(#[from] cosmwasm_std::OverflowError),
```

Comment:
when does this happen? is this related to the `checked_add`? doesnt that return None of overflows?

Reply:
This can happen with any integer type in Rust. In debug builds overflow will trigger a panic, unless the add is checked. Release won't panic, but will do something (overflow, something else, etc). Since this is using cosmwasm types and not native Rust types, I'm not sure what the behavior will be, but if there's overflow (unlikely, since it's 128 bits), it probably should be treated as an error. The alternative is to use a saturating add.